test/helper.rb

```diff
@@ -87,17 +84,26 @@ def self.call(host = '127.0.0.1')
 
 require "timeout"
 
-class TimeoutTestCase < Minitest::Test # rubocop:disable Puma/TestsMustTimeout
+module TimeoutPrepend
+
+  TEST_CASE_TIMEOUT = ENV.fetch("TEST_CASE_TIMEOUT") do
+    RUBY_ENGINE == "ruby" ? 45 : 60
+  end.to_i
+
   # our own subclass so we never confuse different timeouts
   class TestTookTooLong < Timeout::Error
   end
 
-  def self.run_one_method(klass, method_name, reporter)
-    ::Timeout.timeout($test_case_timeout, TestTookTooLong) do
-      super
+  def capture_exceptions(&blk)
+    super do
+      ::Timeout.timeout(TEST_CASE_TIMEOUT, TestTookTooLong) { yield }
     end
   end
+end
+
+Minitest::Test.prepend TimeoutPrepend
 
+class TimeoutTestCase < Minitest::Test # rubocop:disable Puma/TestsMustTimeout
```

Comment:
Todo, can be separate PR: we should just rename this to PumaTest or something, since it doesn't have timeout responsibilities anymore.